### PR TITLE
Update go sample with TWData/TWString

### PR DESCRIPTION
**Pull request by hewigovens**

### CODE REVIEW

**Comment by hewigovens** [comment]
codacy GoLint_GoLint/Comments/DocComments disabled

File: samples/go/main.go [hunk 1]
```diff
@@ -4,14 +4,66 @@ package main
 // #cgo LDFLAGS: -L/wallet-core/build -L/wallet-core/build/trezor-crypto -lTrustWalletCore -lprotobuf -lTrezorCrypto -lc++ -lm
 // #include <TrustWalletCore/TWHDWallet.h>
 // #include <TrustWalletCore/TWString.h>
+// #include <TrustWalletCore/TWData.h>
```
Comment (madcake) [inline samples/go/main.go]: Cold we remove these lines?

File: samples/go/main.go [hunk 2]
```diff
@@ -4,14 +4,66 @@ package main
...
+// Go string -> C.TWString
+func TWStringCreateWithGoString(s string) unsafe.Pointer {
+    cStr := C.CString(s)
+    defer C.free(unsafe.Pointer(cStr))
+    str := C.TWStringCreateWithUTF8Bytes(cStr)
+    return str
+}
+
+// C.TWData -> Go byte[]
+func TWDataGoBytes(d unsafe.Pointer) []byte {
```
Comment (madcake) [inline samples/go/main.go]: Could we add codegen for it?

File: samples/go/main.go (hunk 1 above)
Comment (hewigovens) [reply samples/go/main.go]: no, these are cgo formats

File: samples/go/main.go (hunk 2 above)
Comment (hewigovens) [reply samples/go/main.go]: Yes, these two helpers must be native, we have similar helpers for Java/Objc, we can add go binding to codegen later (if we really need it)

**Pull request merged**